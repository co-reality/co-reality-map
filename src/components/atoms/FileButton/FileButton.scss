@import "scss/constants";

$label-margin: $spacing--sm 0 $spacing--sm 0;

.FileButton {
  display: flex;
  margin-bottom: 3rem;
  flex-direction: column;

  .btn {
    padding: 0px;
  }

  &__label {
    width: 100%;
    cursor: pointer;
    margin: $label-margin;
  }

  &__description {
    color: $light-grey--lightest;
  }
}
