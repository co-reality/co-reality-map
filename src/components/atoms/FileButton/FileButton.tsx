import React, { useCallback } from "react";
import classNames from "classnames";

import "./FileButton.scss";

export interface FileButtonProps {
  title: string;
  description?: string;
  disabled?: boolean;
  onChange: (url: string, file: FileList) => void;
}

export const FileButton: React.FC<FileButtonProps> = ({
  title,
  description,
  disabled,
  onChange,
}) => {
  const handleChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      const files = event.target.files;

      if (!files || disabled) return;

      const url = URL.createObjectURL(files[0]);

      onChange(url, files);
    },
    [disabled, onChange]
  );

  const buttonClasses = classNames("btn btn-primary", {
    "btn-disabled": disabled,
  });

  return (
    <div className="FileButton">
      <button className={buttonClasses} disabled={disabled}>
        <label className="FileButton__label" htmlFor="fileButton">
          {title}
        </label>
      </button>

      <input
        hidden
        type="file"
        id="fileButton"
        disabled={disabled}
        onChange={handleChange}
      />

      {description && (
        <span className="FileButton__description">{description}</span>
      )}
    </div>
  );
};
