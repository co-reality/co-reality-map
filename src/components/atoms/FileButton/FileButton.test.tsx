import React from "react";
import renderer from "react-test-renderer";

// Test component
import FileButton from "./FileButton";

describe("<FileButton />", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<FileButton onChange={() => {}} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
