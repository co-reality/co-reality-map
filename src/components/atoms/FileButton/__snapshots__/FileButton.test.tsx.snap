// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileButton /> renders correctly 1`] = `
<div
  className="sc-gsTCUz bsXUus"
>
  <label
    className="sc-bdfBwQ sc-dlfnbm eqEdkr cbbQCX"
    htmlFor="backgroundMap"
  >
    Import a map background
  </label>
  <input
    hidden={true}
    id="backgroundMap"
    onChange={[Function]}
    type="file"
  />
  <span
    className="sc-hKgILt fwiLOX"
  >
    Recommended size: 2000px / 1200px
  </span>
</div>
`;
